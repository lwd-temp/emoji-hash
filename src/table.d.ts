declare module 'table.js' {
  const emojiTable: { [key: string]: string };
  export default emojiTable;
}
