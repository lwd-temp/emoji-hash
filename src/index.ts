import transformBinary from './methods/transformBinary';
import getBitwise from './methods/getBitwise';

function getHash(input: string): string {
  return transformBinary(getBitwise(input));
}

export default getHash;
