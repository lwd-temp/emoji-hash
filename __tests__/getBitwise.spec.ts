import { getBitwise } from '../src';
import { describe, test } from '@jest/globals';

const testTableForHash = [
  'https://yandex.ru',
  'https://github.com/opa-oz/emoji-hash',
  'My text is awesome',
  'Hi guys',
  'opa-oz',
  '@gahara',
  'Python',
  "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book",
  ''
];

describe('Get bitwise of string', () => {
  testTableForHash.forEach((input) => {
    test('should generate hash correctly', () => {
      expect(getBitwise(input)).toMatchSnapshot();
    });
  });
});
