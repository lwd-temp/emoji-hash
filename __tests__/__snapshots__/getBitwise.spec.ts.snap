// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get bitwise of string should generate hash correctly 1`] = `1915103089`;

exports[`Get bitwise of string should generate hash correctly 2`] = `-633154078`;

exports[`Get bitwise of string should generate hash correctly 3`] = `-773262868`;

exports[`Get bitwise of string should generate hash correctly 4`] = `-1780413401`;

exports[`Get bitwise of string should generate hash correctly 5`] = `-1010760648`;

exports[`Get bitwise of string should generate hash correctly 6`] = `-287727358`;

exports[`Get bitwise of string should generate hash correctly 7`] = `-1889329924`;

exports[`Get bitwise of string should generate hash correctly 8`] = `-1512942320`;

exports[`Get bitwise of string should generate hash correctly 9`] = `0`;
